newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -137,219 +69,192 @@ type State = {|
+      // Empty the list if there are only dividers inside it
+      if (optionList.length === separatorCount) optionList = [];
+
+      // Remove divider(s) if they are at the start or end of array
+      while (
+        optionList[optionList.length - 1] !== undefined &&
+        optionList[optionList.length - 1].type !== undefined
+      )
+        optionList.pop();
+      while (
+        optionList[optionList.length - 1] !== undefined &&
+        optionList[0].type !== undefined
+      )
+        optionList.shift();
+
+      return optionList;
+    };
+
+    const helperText = this.props.helperMarkdownText ? (
+      <MarkdownText source={this.props.helperMarkdownText} />
     ) : null;
 
+    const getOptionLabel = (option: Option): string => {
+      // option.value ? option.value : getCurrentInputValue();
+      // Writing shorthand will fail flow checks.
+      // Explicitly define all the cases of type Option.
+      if (option.value) return option.value;
+      else return getCurrentInputValue();
+    };
+
+    const handleChange = (event, option: Option | string): void => {
+      // Explicitly handle option.type to avoid flow errors
+      if (option !== null && option.type !== 'separator') {
+        if (typeof option === 'string') {
+          this.props.onChange(option);
+        } else {
+          if (option.onClick) option.onClick();
+          else {
+            if (this.props.onChoose) this.props.onChoose(option.value);
+            else this.props.onChange(option.value);
+          }
+        }
+      }
+      if (typeof event.key === 'string') handleKeyDown(event.key);
+      if (this.props.onRequestClose) this.props.onRequestClose();
+    };
+
+    const handleKeyDown = (key: string): void => {
+      if (key === 'Enter' && this._input.current !== null)
+        this._input.current.blur();
+    };
+
+    const handleInputChange = (e, value: string, reason: string): void =>
+      this.setState({ inputValue: value });
+
+    const getDefaultStylingProps = (params: Object): Object => {
+      var { InputProps, inputProps, InputLabelProps, ...other } = params;
+      InputProps.className = null;
+      InputProps.endAdornment = null;
+      inputProps.className = null;
+      InputProps.style = styles.inputRoot;
+      inputProps.style = styles.inputInput;
+      inputProps = {
+        ...inputProps,
+        onKeyDown: (e): void => {
+          if (e.key === 'Escape' && this.props.onRequestClose)
+            this.props.onRequestClose();
+        },
+      };
+      return { InputProps, inputProps, InputLabelProps, ...other };
+    };
+
+    const renderItem = (option: Option, state: Object): React.Node => {
+      if (option.type && option.type === 'separator') {
+        return (
+          <ListItem
+            divider
+            disableGutters
+            component={'div'}
+            style={{ padding: 0, margin: 0 }}
+          />
+        );
+      } else {
```

Comment:
You've used a return in the if, so it's fine I think to avoid the else and the extra indentation. it also emphasize that it's the most common path used in the code (you have few separators, but surely a lot of traditional items)